scripts/upsNode.py

```diff
@@ -2,34 +2,36 @@
 
 import PyNUT
 import rospy
-from sensor_msgs.msg import BatteryState
+from robosub.msg import BatteryDetailed
 
 def upsDataToChargeStatus(status):
-    if "CHRG" in status:
-        return BatteryState.POWER_SUPPLY_STATUS_CHARGING
-    if "DISCHRG" in status:
-        return BatteryState.POWER_SUPPLY_STATUS_DISCHARGING
+    statuses = status.split()  # Split on spaces
+    ret = []
+
+    for i in statuses:
+        if "DISCHRG" in i:
+            ret.append(BatteryDetailed.POWER_SUPPLY_STATUS_DISCHARGING)
+        elif "CHRG" in i:
+            ret.append(BatteryDetailed.POWER_SUPPLY_STATUS_CHARGING)
+        elif "OL" in i:
+            ret.append(BatteryDetailed.POWER_SUPPLY_STATUS_ON_LINE)
+        elif "OB" in i:
+            ret.append(BatteryDetailed.POWER_SUPPLY_STATUS_ON_BATTERY)
+        elif "RB" in i:
+            ret.append(BatteryDetailed.POWER_SUPPLY_STATUS_REPLACE)
+        elif "LB" in i:
+            ret.append(BatteryDetailed.POWER_SUPPLY_STATUS_LOW_BATTERY)
+        elif "HB" in i:
+            ret.append(BatteryDetailed.POWER_SUPPLY_STATUS_HIGH_BATTERY)
+        elif "OVER" in i:
+            ret.append(BatteryDetailed.POWER_SUPPLY_STATUS_OVERLOAD)
```

Comment:
`FULL` is missing from this list, yet present in the `echo_battery` list. Is it needed/attainable?